src/lib/api.ts

```diff
@@ -34,6 +33,35 @@ function parseLibraryId(libraryId: string): {
   };
 }
 
+/**
+ * Generates appropriate error messages based on HTTP status codes
+ * @param errorCode The HTTP error status code
+ * @param apiKey Optional API key (used for rate limit message)
+ * @param docMode Optional documentation mode for mode-aware messages
+ * @returns Error message string
+ */
+function createErrorMessage(
+  errorCode: number,
+  apiKey?: string,
+  docMode?: DocumentationMode
+): string {
+  switch (errorCode) {
+    case 429:
+      return apiKey
+        ? "Rate limited due to too many requests. Please try again later."
+        : "Rate limited due to too many requests. You can create a free API key at https://context7.com/dashboard for higher rate limits.";
+    case 404:
+      return "The library you are trying to access does not exist. Please try with a different library ID.";
+    case 401:
+      return `Unauthorized. Please check your API key. The API key you provided (possibly incorrect) is: ${apiKey}. API keys should start with 'ctx7sk'`;
```

Comment:
```suggestion
      return `Unauthorized. Please check your API key. API keys should start with 'ctx7sk'`;
```
let's not include the user's token in the error message, they might render this error somewhere for their users etc.